### ISSUE TITLE
Explore Heroku for arbitrary demo/PR deployments for bedrock

### ISSUE THREAD

**Issue by pmac** [mozmeao/infra#1164]
I've been wondering if Heroku's Review Apps might be a good solution for our ephemeral work-in-progress demos that we've enjoyed via the legacy Deis Workflow deployments. Heroku will now use our [dockerfile to build and deploy the app](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml#review-apps-and-app-json), so very minimal changes are required to make this work. I have an [example app](https://bedrock-demo-pmac.herokuapp.com/en-US/) running and auto-deploying from a [branch in my fork of bedrock](https://github.com/pmac/bedrock/tree/heroku). If this works well then my suggestion for #981 would be to setup our new pipeline for dev, stage, prod, and a few named demos (e.g. demo-1.allizom.org), and leave the rest to Heroku integration using Mozilla's corporate enterprise account.

**Comment by pmac** [comment]
@mozmeao/theconn thoughts?

**Comment by metadave** [comment]
💯 I think this is a great solution to make some of the infra around demos easier to work with. 

Do we need to do anything with certs or DNS?

We'll need a tiny doc somewhere so folks know how to use this feature.

**Comment by safwanrahman** [comment]
I think dokku will be better choice for this. You can host it yourself in mozmeao vm

**Comment by metadave** [comment]
>  You can host it yourself in mozmeao vm

given that out team is small, having someone else host strategic pieces of our infra for us is an advantage.

**Comment by safwanrahman** [comment]
Got it. I thought heroku is expensive so better to host something own to reduce cost. But maybe the human cost will be higher!

**Comment by pmac** [comment]
Right, human costs can be quite high, and since these bedrock instances will be purely for demo and see almost no traffic and require no external services (sqlite database etc.) I think we'll be able to get by for very cheap or free.

**Comment by pmac** [comment]
Though I will say that I use and love Dokku for some personal projects. Fantastic bit of software.

**Comment by pmac** [comment]
I have an auto-deploying Heroku app setup at https://bedrock-demos.herokuapp.com/en-US/. It's hooked up to the `mozilla/bedrock` repo and deploying from the `heroku` branch (for now, until the PR I'll soon file merged).

**Comment by pmac** [comment]
Things are all set and review apps are on. We'll see how it goes. I'll inform the websites team of the new thing.
